scala/private/phases/phase_collect_jars.bzl

```diff
@@ -46,41 +41,32 @@ def phase_collect_jars_junit_test(ctx, p):
     )
     return _phase_collect_jars_default(ctx, p, args)
 
-def phase_collect_jars_library_for_plugin_bootstrapping(ctx, p):
-    args = struct(
-        unused_dependency_checker_mode = "off",
-    )
-    return _phase_collect_jars_default(ctx, p, args)
-
 def phase_collect_jars_common(ctx, p):
     return _phase_collect_jars_default(ctx, p)
 
 def _phase_collect_jars_default(ctx, p, _args = struct()):
     return _phase_collect_jars(
-        ctx,
-        _args.base_classpath if hasattr(_args, "base_classpath") else p.scalac_provider.default_classpath,
-        _args.extra_deps if hasattr(_args, "extra_deps") else [],
-        _args.extra_runtime_deps if hasattr(_args, "extra_runtime_deps") else [],
-        _args.unused_dependency_checker_mode if hasattr(_args, "unused_dependency_checker_mode") else p.unused_deps_checker,
+        ctx = ctx,
+        p = p,
+        base_classpath = _args.base_classpath if hasattr(_args, "base_classpath") else p.scalac_provider.default_classpath,
+        extra_deps = _args.extra_deps if hasattr(_args, "extra_deps") else [],
+        extra_runtime_deps = _args.extra_runtime_deps if hasattr(_args, "extra_runtime_deps") else [],
```

Comment:
nit: I can see most of the cases in this repo, we are not calling functions by keywords. It would be good to make it consistent (I won't point to all the places to bombard the PR). WDYT?

Reply:
I think there were already inconsistencies in the past, and the benefit of keywords is that when you add and remove arguments you don't accidentally end up mixing up arguments between each other. I strongly would prefer keeping them to avoid the future confusion but if the collective will of the maintainers is to drop them then will drop them.

Reply:
The reason is valid. I am also in favor of it though I wan't aware of it before. However, all the current phases are not using keywords ATM, I prefer to have a separate PR to make things consistent. Let's wait for suggestion from @ittaiz

Reply:
I think this generally sticks to changing things where we are already changing things; if we do want consistency then I feel like doing the rest in a separate diff (but leaving such changes already in this one here) makes sense. But will defer to @ittaiz .

Reply:
I think consistency has value but simplicity also has its value.
If the number of arguments is small and the types differ enough then I'd probably not use keywords. 
Having said that I definitely agree that there are cases where keywords bring a lot of value.
@Jamie5 can you give a few examples (from the phases files) that were using named arguments?
I'm not against this pattern but I'm a bit concerned it's going to add a lot of verbosity

Reply:
https://github.com/bazelbuild/rules_scala/blob/master/scala/private/phases/phase_compile.bzl#L232 and https://github.com/bazelbuild/rules_scala/blob/master/scala/private/phases/phase_compile.bzl#L413 for example

Yeah for just a few arguments it doesn't seem worthwhile but as the arg count increases it becomes riskier and also more annoying when reading the code and needing to match up arguments.

Reply:
There is a trade-off here between simplicity, safety and consistency.
I'm not sure I'd like all method calls to have named arguments.
If the number of arguments is small and the types differ enough then I'd probably not use keywords. 
Having said that I definitely agree that there are cases where keywords bring a lot of value.
@Jamie5 can you stick to the current convention (of phases at least) and maybe send a separate PR only about adding named arguments? My reasoning is that I'm somewhat partial on this change and I'd rather not block this PR on this decision but at the same time I'm not saying this is completely wrong.

Reply:
Mainly I'm concerned about verbosity and how easy it will be to get inconsistent.
Tried searching for advice in PEP8 but couldn't find any.
If you can find references that talk about one way or another as the default I'd love to see

Reply:
@ittaiz do you mean everywhere this adds/introduces keyword arguments, in `*/phases/*`, to remove them? I am fine with doing so and would probably let it lie as opposed to pushing for it in the future. For me the links due point to named args.

Reply:
I missed this thread. If it's not too much trouble I'd appreciate that you indeed remove those

Reply:
Ok will do